### PR TITLE
Delete validation

### LINKED ISSUE #184: After deleting user, activity still shows as 1 participant.
After the deletion of a user, the number of participants in that activity does not update.

<hr><sub>[original: dawo5010/ped#17]<br/>
</sub>

### PR DESCRIPTION
**Pull request by podocarp**
** (fix #184)**

### CODE REVIEW

**Review by liakify — DISMISSED** [review]
Should be good to go
edit: ignore I missed out something

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 1]
```diff
@@ -69,6 +70,19 @@ public void execute_validIndexFilteredList_success() {
...
+    @Test
+    public void execute_deletePersonInActivity_throwsCommandException() {
+        showPersonAtIndex(model, INDEX_FIRST);
+        DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST);
+        Person personToDelete = model.getFilteredPersonList().get(INDEX_FIRST.getZeroBased());
+        String errmsg = String.format(DeleteCommand.MESSAGE_PERSON_INVOLVED_ACTIVITY,
+                personToDelete.getName());
+
+        Model tempmodel = new ModelManager(
+                getTypicalAddressBook(), new UserPrefs(), new InternalState(), getTypicalActivityBook());
```
Comment (liakify) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:122]: I think this line should come first, then everything else executes off `tempmodel`?

Comment (daekoon) [reply src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:122]: No, its fine. The command gets executed only on the `assertCommandFailure` step

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 2]
```diff
@@ -69,6 +70,19 @@ public void execute_validIndexFilteredList_success() {
...
+
+        assertCommandFailure(deleteCommand, tempmodel, errmsg);
+    }
```
Comment (daekoon) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:125]: Can we create one more test for a success case? Add activities with people inside, but make sure that the person being deleted is not inside. Then, assert that the command successfully deletes the contact.

File: src/main/java/seedu/address/model/ActivityBook.java [hunk 3]
```diff
@@ -118,6 +118,20 @@ public boolean equals(Object other) {
         return true;
     }
 
+    /**
+     * Checks whether the person with ID is present in any activity.
+     * @param personId Id of the person to check.
+     * @return True if person exists, false otherwise.
+     */
+    public boolean hasPerson(Integer personId) {
```
Comment (Aulud) [inline src/main/java/seedu/address/model/ActivityBook.java:126]: Nice, we actually need this.

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java (hunk 2 above)
Comment (Aulud) [reply src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:125]: Agree with @daekoon.

**Review by Aulud — DISMISSED** [review]
Feature-wise, it's pretty much good to go, although I would agree with Daekoon's recommendation to add more tests. You could consider:

(1) Creating an activity with A, B, C but then deleting D -> success
(2) Creating an activity with A, B, C, then uninviting C, then deleting C -> success

**Review by Aulud — APPROVED** [review]
Good to go. @liakify Thanks for fixing the side-effect from the way we set up tests in `DeleteCommandTest`. We might need to comb through the rest of the test files subsequently to ensure the other tests are all independent with no side-effects.

**Pull request merged**